### ISSUE TITLE
Mob fan not detecting upgrades properly when unloaded and reloaded?

### ISSUE THREAD

**Issue by Rashkavar** [vadis365/Mob-Grinding-Utils#33]
I've come across what seems to be an intermittant issue with the Mob Fans.  In my Skyfactory 3 single player world (Minecraft 1.10.2, SF3 3.0.8, Mob Grinding Utils 0.1.8.16) I built my mob farm a little ways off from my base - far enough that I'm pretty sure it's out of spawn chunks, since such builds can be lag producers.

After returning from a resupply run to my main base, I flicked on the fans to clear out the mobs that had spawned (being dumb, I built the platforms before installing the mob grinders) so I could place temporary torches, and noticed the mobs at one of the fans were only being pushed to about the halfway point.

After checking to ensure the range upgrades were all indeed installed, and logging out and back in, the fan was still only pushing part way.  I switched into creative to do some troubleshooting and flew into the farm and could also fly well onto the spawning platform before hitting the fan's field of influence.  When testing this, I noticed a second of the 4 fans was also acting up; mobs were stopping 2 blocks shy of the gap, which means it was reading 8 of the 10 range upgrades.  The width and height upgrades were both working fine - blowing on all 3 levels of the farm and effecting the full 28 block width of the platform.  I even tried breaking and re-placing the fan, to no avail.

I went back to base to build some more fans to place above and below the ones that weren't working (the farm design is such that its position is not changable unless you mirror it, and when I returned to the build site, the fans were working properly.

Normally I'd suspect a chunk boundary issue, but the 2 fans that malfunctioned encountered their difficulties only a few blocks apart - if one error was on a chunk boundary, the other most certainly was not.  

And as the error unexpectedly resolved itself, I can no longer provide the save file I had with the error active.  Sorry!

**Comment by vadis365** [comment]
Hello,
I'm not sure what is going on here for you. One thing to note that if any blocks are in front of the fan in a direct line the fan will stop at that block. Any block will stop the fan, a torch, fluids a slab etc.
I added this so it's possible to place a block down in the 1 block line in front of the fan (up to it's full distance) in case you fall in a mob farm or need to approach an active fan.
Cheers,
Me.

**Comment by Rashkavar** [comment]
Hmm. I think that was before I started tinkering with the addition of witch water to the system, which would be the only non-air block that was in there at the fan level.

Good to know about the non-air blocks blocking the fan effect, in case I ever wind up trapping myself with a fan without op powers to throw myself into spectator.  

The issue hasn't cropped up since, except where I've added witchwater, which is triggering that fan range behavior you mentioned, and since that was happening reliably I just added fans to redirect the mobs once they hit the witch water to hit the middle two fans (which are unobstructed).  Good to know it's a deliberate mechanic.  

My guess is that I had some ghost blocks in there or something - I've had weird issues with non-existent light sources that go away when relogging too, so it's possible that kind of issue was at play.

Thanks!
